flocker/node/agents/ebs.py

```diff
@@ -244,6 +256,22 @@ class AttachFailed(Exception):
     """
 
 
+class InvalidStateException(Exception):
+    """
+    A volume is not in an appropriate state to perform an operation.
+
+    :param boto3.resources.factory.ec2.Volume volume: The volume.
+    :param str state: The known volume state at the time of the exception.
+    :param list valid_states: A ``list`` of ``str`` representing states
+        that would have been valid for this operation.
+    """
+    def __init__(self, volume, state, valid_states):
+        Exception.__init__(self)
```

Comment:
Please do `Exception.__init__(self, volume, state, valid_states)` so that the `reason` field in Eliot logs has this information in it.